### PR TITLE
storaged: Make sure to use the D-Bus object manager of Storaged

### PR DESCRIPTION
**Pull request by mvollmer**
By watching for objects on the root path.

### CODE REVIEW

**Comment by mvollmer** [comment]
This still leads to a lot of unexpected calls to Introspect.  Also, GetManagedObjects is called once for each call to client.proxies().

**Comment by stefwalter** [comment]
> This still leads to a lot of unexpected calls to Introspect. 

Not more than before, right? I believe this is a separate issue.

> Also, GetManagedObjects is called once for each call to client.proxies().

Yes, that's expected. We only track watched objects, when the watch changes, we call GetManagedObjects() again. That was the behavior before this change as well.

**Comment by stefwalter** [comment]
Marking priority, so we can enable G_MESSAGES_DEBUG for cockpit-bridge and start to debug some of the other known issues.

**Comment by stefwalter** [comment]
We could add an options: { watch: false } to client.proxies() just like client.proxy() has. That way a caller can watch once and then create the host of proxies lists necessary.

**Comment by mvollmer** [comment]
>> This still leads to a lot of unexpected calls to Introspect. 

> Not more than before, right? I believe this is a separate issue.

Correct, not more than before.  I was expecting almost no calls to Introspect, but maybe I misunderstand.

**Comment by mvollmer** [comment]
> We could add an options: { watch: false } to client.proxies() just like client.proxy() has. That way a caller can watch once and then create the host of proxies lists necessary.

I have done this and it works well for client.proxies().  However, I still need to use client.proxy with 'watch: true.  Is that expected?

**Comment by stefwalter** [comment]
> I have done this and it works well for client.proxies(). However, I still need to use client.proxy with 'watch: true. Is that expected?

Only if your proxy's path is not covered by the previous watch.

File: pkg/storaged/client.js [hunk 1]
```diff
@@ -92,51 +92,48 @@ define([
...
-    client.blocks_fsys = client.storaged_client.proxies("org.storaged.Storaged.Filesystem",
-                                                        "/org/storaged/Storaged/block_devices");
-    client.blocks_crypto = client.storaged_client.proxies("org.storaged.Storaged.Encrypted",
-                                                          "/org/storaged/Storaged/block_devices");
-
-    client.storaged_jobs = client.storaged_client.proxies("org.storaged.Storaged.Job",
-                                                          "/org/storaged/Storaged/jobs");
+
+    function proxy(iface, path) {
+        return client.storaged_client.proxy(iface, path, { watch: true });
```
Comment (stefwalter) [inline pkg/storaged/client.js]: watch: true is the default. I don't think you need to make a special function for this.

Comment (mvollmer) [inline pkg/storaged/client.js]: No, but I would like to use 'watch': false here, so I spelled this out explicitly.

**Comment by mvollmer** [comment]
>> I have done this and it works well for client.proxies(). However, I still need to use client.proxy with 'watch: true. Is that expected?

> Only if your proxy's path is not covered by the previous watch.

I think it should be covered:

```
client.storaged_client.watch({ path_namespace: "/org/storaged/Storaged" });
...
client.manager = proxy("org.storaged.Storaged.Manager", "/org/storaged/Storaged/Manager");
```

**Pull request closed without merging**